### DISCUSSION TITLE
How to reduce the usage of javascript?

### DISCUSSION THREAD

**Discussion by SandeepGumaste** [gatsbyjs/gatsby#34211] [Help]
I'm able to visualize the size of the js files using webpack bundle analyzer, but how to get started on removing the unused js?

**Comment by graysonhicks** [comment]
Hi @SandeepGumaste that depends on what bundles you are wanting to trim down as each is generated differently.

**Reply by SandeepGumaste** [reply]
Hi @graysonhicks , thank you for replying! I was looking to cut down on 2 files : 
1. bootstrap.bundle.min.js, 
2. jquery-3.5.1.slim.min.js

I was not able to figure out how to get started with it

**Reply by graysonhicks** [reply]
Okay.  Neither of those libraries is required by on controlled by Gatsby.  You would need to find where in your site you are using them and remove them.

**Reply by SandeepGumaste** [reply]
But I'm importing some functions from those files and only 20 percent of those files is used. Rest 80 perc is unused.

**Reply by graysonhicks** [reply]
Then it sounds like those libraries don't support tree-shaking when being bundled.
